File: iam-policy-lens/scripts/go/gapic.go
package main

import (
	"strings"
)

type GapicCall struct {
	FullName    string           `json:"fullname"`
	FilePath    string           `json:"file_path"`
	Line        int              `json:"line"`
	Source      string           `json:"source_line"`
	Resolution  string           `json:"resolution"`
	Credentials *CredentialsInfo `json:"credentials,omitempty"`
}

var imports = []string{
	"cloud.google.com/go",
	"google.golang.org/api",
	"google.golang.org/genai",
	"google.golang.org/adk",
}

func isRelevantPackage(pkgPath string) bool {
	for _, p := range imports {
		if pkgPath == p || strings.HasPrefix(pkgPath, p+"/") {
			return true
		}
	}
	return false
}

File: iam-policy-lens/scripts/python/gapic.py
from dataclasses import dataclass
from typing import Optional
from credentials import CredentialsInfo
from utils import clean_gapic_fqn

@dataclass
class GapicCall:
    fullname: str
    file_path: str
    line: int
    source_line: str
    resolution: str
    client_fullname: Optional[str] = None
    credentials: Optional[CredentialsInfo] = None

IMPORTS = ["google.cloud", "google.genai", "vertexai", "google.adk"]

def isRelevantImport(import_name: Optional[str]) -> bool:
    return bool(import_name and any(imp in import_name for imp in IMPORTS))
